polar-core/src/kb.rs

```diff
@@ -689,12 +692,82 @@ impl KnowledgeBase {
             unreachable!()
         }
     }
+
+    pub fn has_rules(&self) -> bool {
+        !self.rules.is_empty() || !self.rule_prototypes.is_empty()
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use crate::error::*;
+
+    #[test]
+    fn test_add_source_file_validation() {
```

Comment:
Not really related to this PR but noticed we didn't have tests for this

Reply:
Just noticed that you added a test in #1143 @gw000 — I clobbered it when resolving merge conflicts b/c I think that behavior (+ more) is covered in this test.